### DISCUSSION TITLE
Not able to install node modules on windows

### DISCUSSION THREAD

**Discussion by kushalgandhi26** [supabase/supabase#21918] [Contribute to Supabase]
By running npm install. Please help me to find solution.

Output:
PS D:\Projects\supabase> npm install
npm WARN cleanup Failed to remove some directories [
npm WARN cleanup   [
npm WARN cleanup     'D:\\Projects\\supabase\\node_modules\\@cspotcode',
npm WARN cleanup     [Error: EPERM: operation not permitted, rmdir 'D:\Projects\supabase\node_modules\@cspotcode'] {
npm WARN cleanup       errno: -4048,
npm WARN cleanup       code: 'EPERM',
npm WARN cleanup       syscall: 'rmdir',
npm WARN cleanup       path: 'D:\\Projects\\supabase\\node_modules\\@cspotcode'
npm WARN cleanup     }
npm WARN cleanup   ],
npm WARN cleanup   [
npm WARN cleanup     'D:\\Projects\\supabase\\node_modules',
npm WARN cleanup     [Error: EPERM: operation not permitted, rmdir 'D:\Projects\supabase\node_modules\style-dictionary\examples\complete\android\demo\src\androidTest\java\com\amazon\styledictionaryexample'] {
npm WARN cleanup       errno: -4048,
npm WARN cleanup       code: 'EPERM',
npm WARN cleanup       syscall: 'rmdir',
npm WARN cleanup       path: 'D:\\Projects\\supabase\\node_modules\\style-dictionary\\examples\\complete\\android\\demo\\src\\androidTest\\java\\com\\amazon\\styledictionaryexample'
npm WARN cleanup     }
npm WARN cleanup   ]
npm WARN cleanup ]
npm ERR! code 1
npm ERR! path D:\Projects\supabase\node_modules\libpg-query
npm ERR! command failed
npm ERR! command C:\Windows\system32\cmd.exe /d /s /c node-pre-gyp install --fallback-to-build
npm ERR! Failed to execute 'C:\Program Files\nodejs\node.exe C:\Users\kygan\AppData\Roaming\nvm\v18.17.0\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js configure --fallback-to-build --module=D:\Projects\supabase\node_modules\libpg-query\build\Release\queryparser.node --module_name=queryparser --module_path=D:\Projects\supabase\node_modules\libpg-query\build\Release --napi_version=9 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v108' (1)
npm ERR! node-pre-gyp info it worked if it ends with ok
npm ERR! node-pre-gyp info using node-pre-gyp@1.0.11
npm ERR! node-pre-gyp info using node@18.17.0 | win32 | x64
npm ERR! node-pre-gyp info check checked for "D:\Projects\supabase\node_modules\libpg-query\build\Release\queryparser.node" (not found)
npm ERR! node-pre-gyp http GET https://supabase-public-artifacts-bucket.s3.amazonaws.com/libpg-query-node/queryparser-v13.3.1-node-v108-win32-x64.tar.gz
npm ERR! node-pre-gyp ERR! install response status 404 Not Found on https://supabase-public-artifacts-bucket.s3.amazonaws.com/libpg-query-node/queryparser-v13.3.1-node-v108-win32-x64.tar.gz
npm ERR! node-pre-gyp WARN Pre-built binaries not installable for libpg-query@13.3.1 and node@18.17.0 (node-v108 ABI, unknown) (falling back to source compile with node-gyp)     
npm ERR! node-pre-gyp WARN Hit error response status 404 Not Found on https://supabase-public-artifacts-bucket.s3.amazonaws.com/libpg-query-node/queryparser-v13.3.1-node-v108-win32-x64.tar.gz
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@9.3.1
npm ERR! gyp info using node@18.17.0 | win32 | x64
npm ERR! gyp info ok
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@9.3.1
npm ERR! gyp info using node@18.17.0 | win32 | x64
npm ERR! gyp info find Python using Python version 3.12.1 found at "C:\Users\kygan\AppData\Local\Programs\Python\Python312\python.exe"
npm ERR! gyp info find VS using VS2017 (15.9.34407.156) found at:
npm ERR! gyp info find VS "C:\Program Files (x86)\Microsoft Visual Studio\2017\BuildTools"
npm ERR! gyp info find VS run with --verbose for detailed information
npm ERR! gyp info spawn C:\Users\kygan\AppData\Local\Programs\Python\Python312\python.exe
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args   'C:\\Users\\kygan\\AppData\\Roaming\\nvm\\v18.17.0\\node_modules\\npm\\node_modules\\node-gyp\\gyp\\gyp_main.py',
npm ERR! gyp info spawn args   'binding.gyp',
npm ERR! gyp info spawn args   '-f',
npm ERR! gyp info spawn args   'msvs',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   'D:\\Projects\\supabase\\node_modules\\libpg-query\\build\\config.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   'C:\\Users\\kygan\\AppData\\Roaming\\nvm\\v18.17.0\\node_modules\\npm\\node_modules\\node-gyp\\addon.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   'C:\\Users\\kygan\\AppData\\Local\\node-gyp\\Cache\\18.17.0\\include\\node\\common.gypi',
npm ERR! gyp info spawn args   '-Dlibrary=shared_library',
npm ERR! gyp info spawn args   '-Dvisibility=default',
npm ERR! gyp info spawn args   '-Dnode_root_dir=C:\\Users\\kygan\\AppData\\Local\\node-gyp\\Cache\\18.17.0',
npm ERR! gyp info spawn args   '-Dnode_gyp_dir=C:\\Users\\kygan\\AppData\\Roaming\\nvm\\v18.17.0\\node_modules\\npm\\node_modules\\node-gyp',
npm ERR! gyp info spawn args   '-Dnode_lib_file=C:\\\\Users\\\\kygan\\\\AppData\\\\Local\\\\node-gyp\\\\Cache\\\\18.17.0\\\\<(target_arch)\\\\node.lib',
npm ERR! gyp info spawn args   '-Dmodule_root_dir=D:\\Projects\\supabase\\node_modules\\libpg-query',
npm ERR! gyp info spawn args   '-Dnode_engine=v8',
npm ERR! gyp info spawn args   '--depth=.',
npm ERR! gyp info spawn args   '--no-parallel',
npm ERR! gyp info spawn args   '--generator-output',
npm ERR! gyp info spawn args   'D:\\Projects\\supabase\\node_modules\\libpg-query\\build',
npm ERR! gyp info spawn args   '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! Traceback (most recent call last):
npm ERR!   File "C:\Users\kygan\AppData\Roaming\nvm\v18.17.0\node_modules\npm\node_modules\node-gyp\gyp\gyp_main.py", line 42, in <module>
npm ERR!     import gyp  # noqa: E402
npm ERR!     ^^^^^^^^^^
npm ERR!   File "C:\Users\kygan\AppData\Roaming\nvm\v18.17.0\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\__init__.py", line 9, in <module>
npm ERR!     import gyp.input
npm ERR!   File "C:\Users\kygan\AppData\Roaming\nvm\v18.17.0\node_modules\npm\node_modules\node-gyp\gyp\pylib\gyp\input.py", line 19, in <module>
npm ERR!     from distutils.version import StrictVersion
npm ERR! ModuleNotFoundError: No module named 'distutils'
npm ERR! gyp ERR! configure error
npm ERR! gyp ERR! stack Error: `gyp` failed with exit code: 1
npm ERR! gyp ERR! stack     at ChildProcess.onCpExit (C:\Users\kygan\AppData\Roaming\nvm\v18.17.0\node_modules\npm\node_modules\node-gyp\lib\configure.js:325:16)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:514:28)
npm ERR! gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:291:12)
npm ERR! gyp ERR! System Windows_NT 10.0.22631
npm ERR! gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\kygan\\AppData\\Roaming\\nvm\\v18.17.0\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "configure" "--fallback-to-build" "--module=D:\\Projects\\supabase\\node_modules\\libpg-query\\build\\Release\\queryparser.node" "--module_name=queryparser" "--module_path=D:\\Projects\\supabase\\node_modules\\libpg-query\\build\\Release" "--napi_version=9" "--node_abi_napi=napi" "--napi_build_version=0" "--node_napi_label=node-v108"
npm ERR! gyp ERR! cwd D:\Projects\supabase\node_modules\libpg-query
npm ERR! gyp ERR! node -v v18.17.0
npm ERR! gyp ERR! node-gyp -v v9.3.1
npm ERR! gyp ERR! not ok
npm ERR! node-pre-gyp ERR! build error
npm ERR! node-pre-gyp ERR! stack Error: Failed to execute 'C:\Program Files\nodejs\node.exe C:\Users\kygan\AppData\Roaming\nvm\v18.17.0\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js configure --fallback-to-build --module=D:\Projects\supabase\node_modules\libpg-query\build\Release\queryparser.node --module_name=queryparser --module_path=D:\Projects\supabase\node_modules\libpg-query\build\Release --napi_version=9 --node_abi_napi=napi --napi_build_version=0 --node_napi_label=node-v108' (1)
npm ERR! node-pre-gyp ERR! stack     at ChildProcess.<anonymous> (D:\Projects\supabase\node_modules\@mapbox\node-pre-gyp\lib\util\compile.js:89:23)
npm ERR! node-pre-gyp ERR! stack     at ChildProcess.emit (node:events:514:28)
npm ERR! node-pre-gyp ERR! stack     at maybeClose (node:internal/child_process:1091:16)
npm ERR! node-pre-gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:302:5)
npm ERR! node-pre-gyp ERR! System Windows_NT 10.0.22631
npm ERR! node-pre-gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "D:\\Projects\\supabase\\node_modules\\@mapbox\\node-pre-gyp\\bin\\node-pre-gyp" "install" "--fallback-to-build"
npm ERR! node-pre-gyp ERR! cwd D:\Projects\supabase\node_modules\libpg-query
npm ERR! node-pre-gyp ERR! node -v v18.17.0
npm ERR! node-pre-gyp ERR! node-pre-gyp -v v1.0.11
npm ERR! node-pre-gyp ERR! not ok

npm ERR! A complete log of this run can be found in: C:\Users\kygan\AppData\Local\npm-cache\_logs\2024-03-10T00_45_10_243Z-debug-0.log

**Comment by SalmanGits** [comment]
what is your device configuration and cpu usage when you do npm i

**Reply by kushalgandhi26** [reply]
Processor	12th Gen Intel(R) Core(TM) i7-12650H   2.70 GHz
Installed RAM	16.0 GB (15.7 GB usable)
System type	64-bit operating system, x64-based processor

CPU usage: upto 13%

**Reply by tlecheva** [reply]
You should try to npm install supabase in WSL2 Ubuntu 2, not from windows / git-bash.

**Reply by kushalgandhi26** [reply]
Thanks 👍
